Storage/SQL/BrowserTable.swift

```diff
@@ -115,22 +124,45 @@ public class BrowserTable: Table {
         return self.run(db, sql: sql, args: args)
     }
 
-    func create(db: SQLiteDBConnection, version: Int) -> Bool {
-        // We ignore the version.
+    func CreateHistoryTable(version: Int = BrowserTable.DefaultVersion) -> String? {
+        return "CREATE TABLE IF NOT EXISTS \(TableHistory) (" +
+            "id INTEGER PRIMARY KEY AUTOINCREMENT, " +
+            "guid TEXT NOT NULL UNIQUE, " +       // Not null, but the value might be replaced by the server's.
+            "url TEXT UNIQUE, " +                 // May only be null for deleted records.
+            "title TEXT NOT NULL, " +
+            "server_modified INTEGER, " +         // Can be null. Integer milliseconds.
+            "local_modified INTEGER, " +          // Can be null. Client clock. In extremis only.
+            "is_deleted TINYINT NOT NULL, " +     // Boolean. Locally deleted.
+            "should_upload TINYINT NOT NULL, " +  // Boolean. Set when changed or visits added.
+            (version > 5 ? "domainId INTEGER REFERENCES \(TableDomains)(id) ON DELETE CASCADE, " : "") +
+            "CONSTRAINT urlOrDeleted CHECK (url IS NOT NULL OR is_deleted = 1)" +
+            ")"
+    }
 
-        let history =
-        "CREATE TABLE IF NOT EXISTS \(TableHistory) (" +
-        "id INTEGER PRIMARY KEY AUTOINCREMENT, " +
-        "guid TEXT NOT NULL UNIQUE, " +       // Not null, but the value might be replaced by the server's.
-        "url TEXT UNIQUE, " +                 // May only be null for deleted records.
-        "title TEXT NOT NULL, " +
-        "server_modified INTEGER, " +         // Can be null. Integer milliseconds.
-        "local_modified INTEGER, " +          // Can be null. Client clock. In extremis only.
-        "is_deleted TINYINT NOT NULL, " +     // Boolean. Locally deleted.
-        "should_upload TINYINT NOT NULL, " +  // Boolean. Set when changed or visits added.
-        "CONSTRAINT urlOrDeleted CHECK (url IS NOT NULL OR is_deleted = 1)" +
-        ") "
+    func CreateDomainsTable(version: Int = BrowserTable.DefaultVersion) -> String? {
+        if version > 5 {
```

Comment:
I really only put these options in for testing reasons (i.e. you can't test the upgrade from v1 to v10 if you don't have the v1 scehma around somewhere). I really just put the tests in because I was sick of having to manually downgrade/upgrade. Maybe we should punt on that test? Ideally maybe we'd copy these schema's over to the testing files, but in our case that would mean copying BrowserDB over as something like:

class BrowserDBv1 : BrowserDB  {
  override var createTableX: String;
}

That probably means some further refactoring to ensure that everything we want to override is overridable (i.e. the list of expected tables/views for each version for example). Worth it? Or just scrap the test?